### PR TITLE
Add Stable Diffusion OpenVINO example

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add Stable Diffusion OpenVINO example.

Next step: 
- Merge `examples/stable_diffusion` and `examples/directml/stable_diffusion`, or rewrite the README.md to link each other. 
- Double check if the model loader code can be unified.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: examples/stable_diffusion/README.md
Comment (jambayk) [inline examples/stable_diffusion/README.md:1]: why is this file deleted? We still need it as the landing page for stable diffusion example. It is used by the ort team too.

Comment (jambayk) [reply examples/stable_diffusion/README.md:1]: We cannot move the directml example since the directml team has multiple blogs and downstream scripts that rely on the pre-existing feature. I think the best option is to put a link to the openvino workflow in this readme like we do for the directml workflow.

**Review by jambayk — DISMISSED** [review]
Please put back the readme at the root of the `stable_diffusion` folder.

File: examples/stable_diffusion/README.md
Comment (jambayk) [reply examples/stable_diffusion/README.md:1]: It has a separate section for all instructions for cuda ep directly in this readme. We created this readme section after talking with the ort model shipping and demo people. They need a general landing page for stable diffusion that is not inside the directml folder.

File: examples/stable_diffusion/openvino/config_text_encoder.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            "type": "OpenVINOConversion",
+            "config": {
+                "user_script": "user_script.py",
+                "example_input_func": "get_text_encoder_example_input",
+                "input": "get_text_encoder_input_shape",
+                "output_model": "text_encoder"
+            }
+        }
+    },
+    "engine": {
```
[the remark this answers is not in the document]
Comment (jambayk) [reply examples/stable_diffusion/openvino/config_text_encoder.json:29]: not a big issue since it would reuse the cache for each run.

File: examples/stable_diffusion/README.md
Comment (jambayk) [reply examples/stable_diffusion/README.md:1]: Thanks for adding it back!

File: examples/stable_diffusion/openvino/config_text_encoder.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiaoyu-work) [reply examples/stable_diffusion/openvino/config_text_encoder.json:29]: We should refactor our code to decouple ONNXRuntime related code at root level configuration. I'm already working on this. I can add a dummy ep for now.

[the remark this answers is not in the document]
Comment (jambayk) [reply examples/stable_diffusion/openvino/config_text_encoder.json:29]: Thanks, that sounds good to me. I remember you brought up this ort coupling with workflows that does not apply to non onnx models. So, I thought it's easier to add a dummy ep for now.

File: examples/stable_diffusion/openvino/config_unet.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                "example_input_func": "get_unet_example_input",
+                "input": "get_unet_input_shape",
+                "output_model": "unet"
+            }
+        }
+    },
+    "engine": {
+        "search_strategy": false,
+        "evaluate_input_model": false,
+        "host": "local_system",
```
Comment (jambayk) [inline examples/stable_diffusion/openvino/config_unet.json:32]: nit: both the host and local system are redundant since it's the default.

**Review by jambayk — APPROVED** [review]

File: examples/stable_diffusion/openvino/config_unet.json (hunk 2 above)
Comment (xiaoyu-work) [reply examples/stable_diffusion/openvino/config_unet.json:32]: Yeah since this is an example for user, I would like to make it clear.

File: olive/engine/footprint.py [hunk 3]
```diff
@@ -74,6 +74,10 @@ def __init__(
         self.objective_dict = objective_dict or {}
         self.is_marked_pareto_frontier = is_marked_pareto_frontier
 
+    def get_output_model_path(self):
```
Comment (guotuofeng) [inline olive/engine/footprint.py:77]: I don't see where it is used. could you please help clarify the changes?
